src/hb-ot-color-cpal-table.hh

```diff
@@ -151,22 +134,17 @@ struct CPAL
     return numPalettes;
   }
 
-  inline void get_color_record (int palette_index, uint8_t &r, uint8_t &g,
-                                              uint8_t &b, uint8_t &a) const
+  inline uint32_t get_color_record_argb (int palette_index) const
   {
-    // We should check if palette_index is in range as it is not done on COLR sanitization
```

Comment:
Wait wait. It's NEVER ok to have such code committed to HarfBuzz. Even if not hooked up. That's how bugs creep in.
As a minimum there should be a big "XXX" comment, which must be resolved before this code is ever called.  See, I cannot check every single array access later. It's the responsibility of original developer to clearly annotate any access checks that must be done but currently are not.

Also, because colorRecords is a C struct that needs access check (as opposed to an ArrayOf<> template which does the access check for you), it must be renamed to colorRecordsZ to signify that and make code review possible.

Reply:
Lol. I just saw it *is* ArrayOf. So it's safe. colorRecordIndices however must be commented out and name end in X because it's not correctly placed and needs to be accessed through StructAfter<>.

Reply:
Yes, just checked ArrayOf implementation before removing my own comment :)

Reply:
I notice now whole thing should be rewritten.